src/details/ArborX_DetailsLegacy.hpp

```diff
@@ -90,6 +90,18 @@ struct LegacyDefaultCallback
   }
 };
 
+struct LegacyDefaultCallbackWithRank
+{
+  int _rank;
+
+  template <typename Predicate, typename OutputFunctor>
+  KOKKOS_FUNCTION void operator()(Predicate const &, int primitive_index,
```

Comment:
Haven't looked through the rest but my initial reaction is "why is that parameter `int` and not `PairValueIndex`?"

Reply:
Because it's being used with `BVH<MemorySpace>` which will do the callback wrapping on its own.